dashboard/app/controllers/plc/user_course_enrollments_controller.rb

```diff
@@ -16,7 +16,12 @@ def create
     if @user_course_enrollment.save
       redirect_to action: :index
     else
-      redirect_to action: :new
+      @user_course_enrollment = Plc::UserCourseEnrollment.find_by(user: @user_course_enrollment.user, plc_course: @user_course_enrollment.plc_course)
+      if @user_course_enrollment
+        redirect_to action: :index
+      else
+        redirect_to action: :new
+      end
     end
   end
 
```

Comment:
The logic I want to say is "If an object with these params exists, go to it. Otherwise go to the new page." I didn't want to make to trips to the db. is there a better way to do this?